drivers/mpu9250/MPU9250.cpp

```diff
@@ -250,8 +312,26 @@ int MPU9250::mpu9250_deinit()
 	return 0;
 }
 
+int MPU9250::pinThread(int cpu)
```

Comment:
Letting the scheduler manage this stuff is always the best idea, because then improvements there will help everyone. But our approach has to be efficient AND accommodate both NuttX and Linux.

Not saying your code is wrong, because I haven't looked at it yet. Just adding my $0.02.